movieRecomander/backend/src/main/java/com/movierecommender/backend/users/user/AppUserController.java

```diff
@@ -52,11 +46,17 @@ public ResponseEntity<AppUser> read(@PathVariable("id") Long id) {
         return ResponseEntity.ok(foundUser.get());
     }
 
+    @PostMapping(path = "register")
+    @ResponseStatus(code = HttpStatus.CREATED, reason = "CREATED")
+    public void registerUser(@Valid @RequestBody AppUser appUser) {
+        appUserService.addNewAppUser(appUser);
+    }
+
     @PutMapping("/{id}")
     @PreAuthorize("hasAnyRole('ROLE_ADMIN', 'ROLE_USER')")
     @ResponseStatus(code = HttpStatus.NO_CONTENT, reason = "UPDATED")
     public void update(@RequestBody AppUserUpdateModel appUserUpdateModel, @PathVariable Long id) {
-
+        ///appuserModel.isValid()
```

Comment:
asta ar trebui implementata si pusa si in PUT , am inteles ce ai facut ca ai transformat-o in local date direct unde aveai nevoie, dar nu prea e ok --- functia getAge se calculeaza cand deja entity-ul e in baza de date si va da eroare la get daca nu reuseste sa o converteasca, in acel punct baza de date e corupta cu acea informatie, de asta m-am gandit sa ai functia isValid care arunca un BusinessException